pkg/kubelet/container/cache.go

```diff
@@ -120,6 +121,17 @@ func (c *cache) UpdateTime(timestamp time.Time) {
 	}
 }
 
+// GetAll gets all PodStatus in the cache
+func (c *cache) GetAll() []*PodStatus {
+	c.lock.Lock()
```

Comment:
Nit: I believe that we just need a `RLock` and a `RUnlock` here, as we are not performing any updates to the locked data structure.